src/utils/fetchWrapper.js

```diff
@@ -1,31 +1,35 @@
-const fetch = require('node-fetch');
+import fetch from 'cross-fetch';
 
 export const promiseCache = {};
 /**
  * Simple wrapper over fetch, for avoiding duplicate API calls
  * @param url
  */
-const fetchWrapper = (url) => {
+const fetchWrapper = async (url) => {
   if (promiseCache[url]) {
-    return promiseCache[url].then((res) => res.clone());
+    const cachedResponse = await promiseCache[url];
+    return cachedResponse.clone();
   }
 
-  const fetchPromise = fetch(url);
+  try {
+    // Fetch and store the response in cache
+    const fetchPromise = fetch(url).then((res) => {
```

Comment:
if you want to use `await`, you might as well use it here as well :-)

Reply:
Right!